src/Compilers/CSharp/Portable/Symbols/Source/SourceMethodSymbol.cs

```diff
@@ -1586,10 +1586,8 @@ protected void CheckFeatureAvailabilityAndRuntimeSupport(SyntaxNode declarationS
                     Binder.CheckFeatureAvailability(declarationSyntax, MessageID.IDS_DefaultInterfaceImplementation, diagnostics, location);
                 }
 
-                if ((hasBody || IsExtern) && !ContainingAssembly.RuntimeSupportsDefaultInterfaceImplementation)
+                if ((hasBody || IsExtern) && !IsStatic && !ContainingAssembly.RuntimeSupportsDefaultInterfaceImplementation)
                 {
-                    // PROTOTYPE(DefaultInterfaceImplementation): It looks like some flavor of static members was supported by
```

Comment:
It is a part of the current runtime spec. @gafter also thought this would be the right thing to do (https://github.com/dotnet/roslyn/pull/18519#pullrequestreview-31680946).